src/auction.js

```diff
@@ -0,0 +1,433 @@
+events.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+  adjustBids(bid);
+});
+
+function Auction(customPriceBucket, granularity) {
+  var _id = utils.getUniqueIdentifierStr();
+  var _adUnits = [];
+  var _adUnitCodes = [];
+  var _targeting = [];
+  var _bidderRequests = [];
+  var _bidsReceived = [];
+  var _auctionStart;
+  var _auctionId = utils.getUniqueIdentifierStr();
+  var _auctionStatus;
+  var _callback;
+  var _timer;
+
+  this.setAdUnits = (adUnits) => _adUnits = adUnits;
+  this.setAdUnitCodes = (adUnitcodes) => _adUnitCodes = adUnitcodes;
+  this.setTargeting = (targeting) => _targeting = targeting;
+  this.setBidderRequests = (bidderRequests) => _bidderRequests = _bidderRequests.concat(bidderRequests);
+  this.setBidsReceived = (bidsReceived) => _bidsReceived = _bidsReceived.concat(bidsReceived);
+  this.setAuctionId = (auctionId) => _auctionId = auctionId;
+  this.setAuctionStart = (auctionStart) => _auctionStart = auctionStart;
+  this.setAuctionStatus = (auctionStatus) => _auctionStatus = auctionStatus;
+  this.setCallback = (callback) => _callback = callback;
+  this.setTimer = (timer) => _timer = timer;
+
+  this.getId = () => _id;
+  this.getAdUnits = () => _adUnits;
+  this.getAdUnitCodes = () => _adUnitCodes;
+  this.getTargeting = () => _targeting;
+  this.getBidderRequests = () => _bidderRequests;
+  this.getBidsReceived = () => _bidsReceived;
+  this.getAuctionStart = () => _auctionStart;
+  this.getAuctionId = () => _auctionId;
+  this.getAuctionStatus = () => _auctionStatus;
+  this.getCallback = () => _callback;
+  this.getTimer = () => _timer;
+
+  this.startAuctionTimer = function(callback, cbtimeout) {
+    this.setCallback(callback);
+    const timedOut = true;
+    const timeoutCallback = this.executeCallback.bind(this, timedOut);
+    let timer = setTimeout(timeoutCallback, cbtimeout);
+    this.setTimer(timer);
+  }
+
+  this.executeCallback = function(timedOut, cleartimer) {
+    // clear timer when done calls executeCallback
+    if (cleartimer) {
+      clearTimeout(this.getTimer());
+    }
+
+    let callback = this.getCallback();
+    if (callback != null) {
+      try {
+        callback.apply($$PREBID_GLOBAL$$);
+      } catch (e) {
+        utils.logError('Error executing bidsBackHandler', null, e);
+      }
+      this.setCallback(null);
+      $$PREBID_GLOBAL$$.clearAuction();
+
+      if (timedOut) {
+        const timedOutBidders = this.getTimedOutBidders();
+
+        if (timedOutBidders.length) {
+          events.emit(CONSTANTS.EVENTS.BID_TIMEOUT, timedOutBidders);
+        }
+      }
+    }
+  }
+
+  this.getTimedOutBidders = function () {
+    return this.getBidsReceived()
+      .map((bidSet) => {
+        return bidSet.bidderCode;
+      })
+      .filter(uniques)
+      .filter(bidder => this.getBidsReceived()
+        .map((bid) => {
+          return bid.bidder;
+        })
+        .filter(uniques)
+        .indexOf(bidder) < 0);
+  };
+
+  this.done = function() {
```

Comment:
This isn't used outside of this file, is it?

If not, it's best not to attach it to the public API. It makes things easier on people who come to your file thinking "how am I supposed to use an Auction?"

The same goes for any other functions which are only for internal use.